addons/api/addon/workers/sqlite-worker.js

```diff
@@ -6,20 +6,149 @@
 import { PWBWorker } from 'promise-worker-bi';
 import sqlite3InitModule from '@sqlite.org/sqlite-wasm';
 import sharedServiceInit from './utils/shared-service';
+import {
+  CLEAR_DB,
+  CREATE_TABLES,
+  DELETE_STATEMENT,
+  INSERT_STATEMENTS,
+} from './utils/schema';
+
+/** @type {import('@sqlite.org/sqlite-wasm').OpfsSAHPoolDatabase} */
+let db;
+/** @type {import('@sqlite.org/sqlite-wasm').SAHPoolUtil} */
+let poolUtil;
+/** @type {import('@sqlite.org/sqlite-wasm').Sqlite3Static} */
+let sqlite3;
+
+// Maximum number of host parameters for SQLite prepared statements.
+// See "Maximum Number Of Host Parameters In A Single SQL Statement" in
+// https://www.sqlite.org/limits.html
+const MAX_HOST_PARAMETERS = 32766;
+const SCHEMA_VERSION = 1;
 
 const methods = {
   initializeSQLite: async () => {
+    sqlite3 = await sqlite3InitModule({
+      print: console.log,
+      printErr: console.error,
+    });
+    console.log(`SQLite Version: ${sqlite3?.version.libVersion}`);
+    const dbName = await promiseWorker.postMessage({
+      method: 'getDatabaseName',
+    });
+
+    let done = false;
+    while (!done) {
+      try {
+        poolUtil = await sqlite3.installOpfsSAHPoolVfs();
+        db = new poolUtil.OpfsSAHPoolDb(dbName);
+
+        const [row] = db.exec('PRAGMA user_version;', { rowMode: 'object' });
+        // Check if we're on a newer schema version and clear the database if so
+        if (SCHEMA_VERSION > row.user_version) {
+          db.exec(CLEAR_DB);
+        }
+
+        db.exec(CREATE_TABLES(SCHEMA_VERSION));
+        done = true;
+      } catch (err) {
+        // If we get a SQLite3Error and the pool is at capacity, we'll assume the error is
+        // because we hit the cap so we add more capacity and see if that solves it.
+        if (
+          err.name === 'SQLite3Error' &&
+          poolUtil.getCapacity() === poolUtil.getFileCount()
```

Comment:
question for understanding: why not check for `getCapacity` >= to `getFileCount` and also move the `getCapacity` > 20 check above this one?

Reply:
You're right, it likely can be moved inside the catch which I did. I still kept it after this if check though because it only needs to be checked _after_ we add capacity and should be the last thing we do since you can't interact anymore after removing the VFS, otherwise you'd need to `continue` the loop early.

Reply:
ah ok, thank for explanation!